### PR TITLE
Anko - Change Tracking Progress From Xml To Anko

### PR DESCRIPTION
**Pull request by at-datbui**
#### Related issues (Required)
- Task #136

#### Description (Required)
- [x] Change tracking progress from xml to anko

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+import android.view.ViewManager
+import android.widget.*
+import com.hypertrack.lib.internal.consumer.utils.AnimationUtils
+import com.hypertrack.lib.internal.consumer.view.CircularSeekBar
+import org.jetbrains.anko.*
+import org.jetbrains.anko.sdk25.coroutines.onClick
+import vn.asiantech.way.R
+import vn.asiantech.way.extension.circularSeekBar
+
+/**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: WARNING: One class should have one responsibility. Large classes tend to handle many things at once. Split up large classes into smaller classes that are easier to understand.

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+ * Copyright © 2017 Asian Tech Co., Ltd.
+ * Created by datbuit. on 30/11/2017.
+ */
+class TrackingProgressInfo(context: Context) : RelativeLayout(context) {
+    companion object {
+        const val ROTATION_VALUE = 90f
+        const val LAYOUT_WEIGHT = 1f
+        const val LEFT_MARGIN_TV = 10.5f
+        const val RIGHT_MARGIN_TV = -5.5f
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+        const val BOTTOM_MARGIN_TV = -4.5f
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 4]
```diff
@@ -25,3 +27,12 @@ internal fun ViewManager.rippleView(init: RippleView.() -> Unit): RippleView {
 internal fun ViewManager.bottomCard(init: BottomButtonCard.() -> Unit): BottomButtonCard {
     return ankoView({ BottomButtonCard(it) }, 0, init)
 }
+
+internal fun ViewManager.trackingProgress(init: TrackingProgressInfo.() -> Unit):
+        TrackingProgressInfo {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 5]
```diff
@@ -25,3 +27,12 @@ internal fun ViewManager.rippleView(init: RippleView.() -> Unit): RippleView {
...
+    return ankoView({ TrackingProgressInfo(it) }, 0, init)
+}
+
+internal fun ViewManager.circularSeekBar(init: CircularSeekBar.() -> Unit): CircularSeekBar {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt (hunk 1 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: WARNING: One class should have one responsibility. Large classes tend to handle many things at once. Split up large classes into smaller classes that are easier to understand.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt (hunk 4 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt (hunk 5 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+    }
+
+    internal lateinit var llTrackingProgress: LinearLayout
+    internal lateinit var llTrackingInfo: LinearLayout
+    internal lateinit var llActionTracking: LinearLayout
+    internal lateinit var llDetailArrived: LinearLayout
+    internal lateinit var rlActionDetail: RelativeLayout
+    internal lateinit var rlCollapse: RelativeLayout
+    internal lateinit var rlExpandedInfo: RelativeLayout
+    internal lateinit var imgBtnResetPosition: ImageView
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: `imgBtnResetPosition` -> `imgResetPosition`

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 7]
```diff
@@ -0,0 +1,449 @@
...
+                                    tvTime = textView(resources.getString(R.string.mask)) {
+                                        textColor = ContextCompat.getColor(context, R.color.tracking_experience)
+                                        textSize = px2dip(dimen(R.dimen.text_base))
+                                    }.lparams(matchParent, wrapContent)
+
+                                    tvDistance = textView(resources.getString(R.string.mask)) {
+                                        gravity = Gravity.CENTER_VERTICAL
+                                        textSize = px2dip(dimen(R.dimen.text_base))
+                                        textColor = ContextCompat.getColor(context, R.color.gray)
+                                    }.lparams(wrapContent, wrapContent) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: if `width = height = wrapContent` you don't need set on `lparams`
and also check all this file

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 8]
```diff
@@ -0,0 +1,449 @@
...
+                                    below(tvActionStatus)
+                                }
+                            }
+
+                            rlExpandedInfo = relativeLayout {
+                                lparams(matchParent, wrapContent) {
+                                    leftMargin = dimen(R.dimen.track_margin_value_low)
+                                    rightMargin = dimen(R.dimen.track_margin_value_low)
+                                    below(R.id.tracking_progress_info_rl_collapse)
+                                }
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: if back ground is a `color resource` please use:
`backgroundResource = R.color.resource_name` also check all this file

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 9]
```diff
@@ -0,0 +1,449 @@
...
+                .lparams(matchParent, wrapContent) {
+                    topMargin = dimen(R.dimen.track_top_margin)
+                }
+    }
+
+    private fun ViewManager.textViewTrackingInfo(value: String, tvId: Int) = textView(value) {
+        textColor = ContextCompat.getColor(context, R.color.colorBlack)
+        id = tvId
+        gravity = Gravity.CENTER
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 10]
```diff
@@ -0,0 +1,449 @@
...
+import android.view.ViewManager
+import android.widget.*
+import com.hypertrack.lib.internal.consumer.utils.AnimationUtils
+import com.hypertrack.lib.internal.consumer.view.CircularSeekBar
+import org.jetbrains.anko.*
+import org.jetbrains.anko.sdk25.coroutines.onClick
+import vn.asiantech.way.R
+import vn.asiantech.way.extension.circularSeekBar
+
+/**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: WARNING: One class should have one responsibility. Large classes tend to handle many things at once. Split up large classes into smaller classes that are easier to understand.

File: app/src/main/java/vn/asiantech/way/extension/Views.kt (hunk 4 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt (hunk 5 above)
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 11]
```diff
@@ -0,0 +1,448 @@
...
+
+                            rlCollapse = relativeLayout {
+                                id = R.id.tracking_progress_info_rl_collapse
+                                lparams {
+                                    rightOf(R.id.tracking_progress_info_circle_progress_bar)
+                                }
+                                onClick {
+                                    if (rlExpandedInfo.visibility == View.GONE) {
+                                        rlExpandedInfo.visibility = View.VISIBLE
+                                        imgArrow.setImageResource(R.drawable.ic_keyboard_arrow_down_black_18dp)
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: Please change to  `mgArrow.imageResource = R.drawable.ic_keyboard_arrow_down_black_18dp` 
also check all file

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 12]
```diff
@@ -0,0 +1,448 @@
...
+                                            resources.getString(R.string.mask),
+                                            resources.getString(R.string.speed))
+                                            .lparams(0, matchParent, LAYOUT_WEIGHT)
+                                    verticalLayout {
+                                        lparams(0, matchParent, LAYOUT_WEIGHT)
+                                        tvBattery = textViewTrackingInfo(resources.getString(R
+                                                .string.mask),
+                                                R.id.tracking_progress_info_tv_battery)
+                                                .lparams(matchParent, wrapContent)
+                                        imageView(R.drawable.ic_battery_icon) {}
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: remove `{}`

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review this PR, please check it again.
thank you!

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 13]
```diff
@@ -25,3 +27,12 @@ internal fun ViewManager.rippleView(init: RippleView.() -> Unit): RippleView {
 internal fun ViewManager.bottomCard(init: BottomButtonCard.() -> Unit): BottomButtonCard {
     return ankoView({ BottomButtonCard(it) }, 0, init)
 }
+
+internal fun ViewManager.trackingProgress(init: TrackingProgressInfo.() -> Unit):
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: Move this ext function into TrackingProgressInfo class

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 14]
```diff
@@ -0,0 +1,454 @@
...
+        return this
+    }
+
+    internal fun hideArrivedInfor(): TrackingProgressInfo {
+        llDetailArrived.visibility = View.VISIBLE
+        return this
+    }
+}
+
+
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 15]
```diff
@@ -0,0 +1,454 @@
...
+internal fun ViewManager.trackingProgress(init: TrackingProgressInfo.() -> Unit):
+        TrackingProgressInfo {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 16]
```diff
@@ -4,9 +4,11 @@ import android.content.Context
 import android.view.View
 import android.view.ViewManager
 import android.view.inputmethod.InputMethodManager
+import com.hypertrack.lib.internal.consumer.view.CircularSeekBar
 import com.hypertrack.lib.internal.consumer.view.RippleView
 import org.jetbrains.anko.custom.ankoView
 import vn.asiantech.way.ui.custom.BottomButtonCard
+import vn.asiantech.way.ui.custom.TrackingProgressInfo
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Unused import

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 17]
```diff
@@ -25,3 +27,7 @@ internal fun ViewManager.rippleView(init: RippleView.() -> Unit): RippleView {
 internal fun ViewManager.bottomCard(init: BottomButtonCard.() -> Unit): BottomButtonCard {
     return ankoView({ BottomButtonCard(it) }, 0, init)
 }
+
+internal fun ViewManager.circularSeekBar(init: CircularSeekBar.() -> Unit): CircularSeekBar {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 18]
```diff
@@ -0,0 +1,453 @@
...
+import android.widget.*
+import com.hypertrack.lib.internal.consumer.utils.AnimationUtils
+import com.hypertrack.lib.internal.consumer.view.CircularSeekBar
+import org.jetbrains.anko.*
+import org.jetbrains.anko.custom.ankoView
+import org.jetbrains.anko.sdk25.coroutines.onClick
+import vn.asiantech.way.R
+import vn.asiantech.way.extension.circularSeekBar
+
+/**
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: WARNING: One class should have one responsibility. Large classes tend to handle many things at once. Split up large classes into smaller classes that are easier to understand.

File: app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt [hunk 19]
```diff
@@ -0,0 +1,453 @@
...
+    }
+
+    internal fun hideArrivedInfor(): TrackingProgressInfo {
+        llDetailArrived.visibility = View.VISIBLE
+        return this
+    }
+}
+
+internal fun ViewManager.trackingProgress(init: TrackingProgressInfo.() -> Unit):
+        TrackingProgressInfo {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/TrackingProgressInfo.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/extension/Views.kt [hunk 20]
```diff
@@ -35,3 +36,7 @@ internal fun Dialog.setDialogScreenSize(context: Context) {
     windowParams.width = width - AppConstants.TYPE_DIALOG_MARGIN_WIDTH
     this.window.attributes = windowParams
 }
+
+internal fun ViewManager.circularSeekBar(init: CircularSeekBar.() -> Unit): CircularSeekBar {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING:

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/extension/Views.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

**Pull request merged**